flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/agg/batch/HashAggCodeGenHelper.scala

```diff
@@ -493,7 +493,7 @@ object HashAggCodeGenHelper {
 
         if (filterArg >= 0) {
           var filterTerm = s"$inputTerm.getBoolean($filterArg)"
-          if (ctx.nullCheck) {
+          if (inputType.isNullable) {
```

Comment:
we need to be careful with these `isNullable` checks. I agree with Francesco but we should avoid introducing bugs for little to no performance optimizations. This check seems wrong to me. We should check the filterArg type not the input type.